src/scrap.py

```diff
@@ -0,0 +1,254 @@
+        link_job = element.find("a", class_="_3t5uN8xUmg0TOwRCOGQEcU")
+
+        job = {}
+
+        if link_job is not None:
+            job["_title"] = header_title.text
+            job["_link"] = link_job.get("href")
+
+            reddit_jobs.append(job)
+
+    return reddit_jobs
+
+
+# -------------------FOR RAPIDAPI----------------------------
+
+
+def rapid_response() -> List[object]:
+    url = os.getenv("RAPID_API_URL")
+
+    rapid_api_key = os.getenv("RAPID_API_KEY")
+    rapid_api_host = os.getenv("RAPID_API_HOST")
+
+    headers = {
+        "X-RapidAPI-Key": rapid_api_key,
+        "X-RapidAPI-Host": rapid_api_host,
+    }
+
+    rapid_jobs = []
+    response = requests.get(url, headers=headers).json()
+
+    for elem in response["hits"]:
+        job = {}
+        job["_company"] = elem["company_name"]
+        job["_title"] = elem["title"]
+        job["_location"] = elem["location"]
+        job["_link"] = f'https://www.indeed.com/viewjob?jk={elem["id"]}&locality=us'
+
+        rapid_jobs.append(job)
+
+    return rapid_jobs
+
+
+# -------------------FOR INDEED API----------------------------
+
+
+# -------------------FOR LINKEDIN----------------------------
+
+
+def linkedin_response() -> List[object]:
+    url = os.getenv("LINKEDIN_URL")
+
+    response = requests.get(url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
+
+    div_post = parse_content.find_all(
+        "div",
+        class_="base-card relative w-full hover:no-underline focus:no-underline base-card--link base-search-card base-search-card--link job-search-card",
+    )
+
+    linked_in_jobs = []
+
+    for elem in div_post:
+        job = {}
+        job["_company"] = elem.find(
+            "h4", class_="base-search-card__subtitle"
+        ).text.strip()
+        job["_title"] = elem.find("h3", class_="base-search-card__title").text.strip()
+        job["_location"] = elem.find(
+            "span", class_="job-search-card__location"
+        ).text.strip()
+        job["_link"] = elem.find("a", class_="base-card__full-link")["href"]
+
+        linked_in_jobs.append(job)
+
+    return linked_in_jobs
+
+
+# -------------------FOR HELPER FUNCTIONS ----------------------------
+
+# List opportunities shows all the opportunities that have NOT been posted yet
+
+
+def list_opportunities() -> List[object]:
+    connection = sqlite3.connect(job_db)
+    cursor = connection.cursor()
+
+    cursor.execute("SELECT * FROM jobs")
+    rows = cursor.fetchall()
+
+    # Printing out for debugging purposes
+    for row in rows:
+        _company, _title, _location, _link, _processed = zip(*row)
+
+        if not _processed:
+            print("Company:", _company)
+            print("Title:", _title)
+            print("Location:", _location)
+            print("Link:", _link)
+            print("Processed:", _processed)
+            print(" ")
+
+    connection.close()
+
+    return rows
```

Comment:
Rename `list_opportunities` to `list_filtered_opportunities`. This indicates that the function also filters out the _processed_ opportunities.

```suggestion
# List opportunities shows all the opportunities that have NOT been posted yet
def list_filterd_opportunities() -> List[object]:
    connection = sqlite3.connect(job_db)
    cursor = connection.cursor()

    cursor.execute("SELECT * FROM jobs WHERE processed = 0")
    rows = cursor.fetchall()

    # Printing out for debugging purposes
    for row in rows:
        company, title, location, link, processed = row
        print("Company:", company)
        print("Title:", title)
        print("Location:", location)
        print("Link:", link)
        print("Processed:", processed)
        print(" ")

    connection.close()

    return rows
```

Reply:
Done with this!